pkg/controller/host/predicate.go

```diff
@@ -90,8 +90,8 @@ func (r *ProviderPredicate) Update(e event.UpdateEvent) bool {
 func (r *ProviderPredicate) Delete(e event.DeleteEvent) bool {
 	p, cast := e.Object.(*api.Provider)
 	if cast {
-		reconciled := p.Status.ObservedGeneration == p.Generation
-		return reconciled
```

Comment:
Noticed this while troubleshooting other issues.  We need to process `Deleted` events regardless of whether the resource is reconciled (or not).

This applies to similar changes elsewhere.